### PR TITLE
Update license headers for 2024

### PR DESCRIPTION
**Pull request by klmcadams**
- Added -i and --black to docformatter. Previously, docformatter didn't change the files when the hook failed, but with -i it does now
- Updated add-license-headers to v0.2.4
- Removed .reuse directory (no longer needed with this version)
- Added 2024 to the license headers

### CODE REVIEW

File: examples/00_additive_single_bead.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
-# Copyright (C) 2023 ANSYS, Inc. and/or its affiliates.
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
```
Comment (pkrull-ansys) [inline examples/00_additive_single_bead.py:1]: @klmcadams I thought the Ansys copyright format required us to only use a single year, not a span. Did legal, or somebody else, change that?

Comment (klmcadams) [reply examples/00_additive_single_bead.py:1]: @pkrull-ansys The .reuse library that the add-license-header hook uses to create the headers makes it a span if there's already an existing year in the copyright statement. I can double check to make sure that's okay

Comment (RobPasMue) [reply examples/00_additive_single_bead.py:1]: We can verify with legal just in case, but it should be allowed.

**Review by RobPasMue — APPROVED** [review]

File: examples/00_additive_single_bead.py (hunk 1 above)
Comment (pkrull-ansys) [reply examples/00_additive_single_bead.py:1]: @klmcadams @RobPasMue This was my bad. I was mistaken about the format, which is defined on pages 31-32 in https://ansys.sharepoint.com/:b:/r/sites/LegalSHARESite/Shared%20Documents/Intellectual%20Property/All%20Employee%20Access/IP%20Handbook%20-%20Current/IP%20Handbook%202021%20update_2022%20Jan%2025.pdf?csf=1&web=1&e=9VYsqT

**Review by pkrull-ansys — APPROVED** [review]

**Comment by pkrull-ansys** [comment]
@klmcadams It looks like your commit wasn't signed. Can you create another PR with a signed commit? The project settings won't allow merging without signed commits.

**Pull request closed without merging**